FastSimulation/Tracking/python/SeedingMigration.py

```diff
@@ -0,0 +1,28 @@
+import FWCore.ParameterSet.Config as cms
+
+def _regionProducerToFactoryPSet(producer):
```

Comment:
I agree it is a bit of a hack, and should be temporary until FastSim is fully migrated (in a way or another, see more discussion below). At least it allows to migrate components in stages (avoiding even larger PRs).

The region part should be straightforward to migrate to use the `EDProducer`s.  

The hit generator is more difficult. AFAIK FastSim wants to call the same hit generator as standard RECO, but using a different interface (a call per hit ntuplet candidate instead of a call for all hits of an event). I don't know how to deal with that nicely.

One option could be to move the hit generator algorithm to a data product, and the EDProducer producing that would be the same between FastSim and RECO. On RECO side we'd then the hit generator `EDProducer`'s of this PR would read the algorithm from event. (I'm not very happy on the affect to RECO).

Another option would be to split the FastSim seeding to multiple `EDProducer`s, but likely even then the same hit generator `EDProducer` could not be used between FastSim and RECO.

Would be good if somebody from FastSim team would take a look and comment.

@slava77 The region part I could do already in this PR (if really want), but the hit generation part I'd leave to be dealt with later.